src/formatting/pairs.rs

```diff
@@ -116,6 +116,19 @@ fn rewrite_pairs_multiline<T: Rewrite>(
 
     result.push_str(&list.list[0].1.as_ref()?);
 
+    /* Determine the expression kind */
+    let (e, _s) = match list.list.first() {
+        Some(x) => x,
+        None => return None,
+    };
+    let list_len = list.list.len();
+    /* ???? Is there better way than transmute() ???? */
+    let list_expr = unsafe { std::mem::transmute::<&T, &ast::Expr>(e) };
```

Comment:
As noted on the other thread, there's a root problem with comments between the subexpr and type in cast expressions which is most certainly impacting the formatting you are seeing with your snippets. 

We really need to focus on the addressing the core issue instead of associated items, especially an approach like this as there are currently no unsafe blocks within the rustfmt source to my knowledge.